sycl/include/CL/sycl/detail/scheduler/commands.cpp

```diff
@@ -14,7 +14,7 @@
 #include <CL/sycl/detail/scheduler/requirements.h>
 #include <CL/sycl/exception.hpp>
 
-#if (defined(__SYCL_VENDOR_XILINX_EXTENSIONS__))
+#if (defined(__SYCL_XILINX_ONLY__) && defined(__cplusplus) && (__cplusplus >= 201703L))
 #include <CL/sycl/xilinx/fpga/kernel_properties.hpp>
```

Comment:
Probably similar comment.
Probably OK for this PR, but we need to think about a future solution at some point.

Reply:
The C++ restriction component can be removed I think, but the __SYCL_XILINX_ONLY__ block should probably remain as its implementation details in this case rather than some user facing piece of code. At least at the moment anyway.